modules/exploits/multi/http/mirth_connect_cve_2023_43208.rb

```diff
@@ -0,0 +1,242 @@
+          target application. The original vulnerability was identified by IHTeam and assigned CVE-2023-37679. Later,
+          researchers from Horizon3.ai determined the patch to be incomplete and published a gadget chain which bypassed
+          the deny list that the original had implemented. This second vulnerability was assigned CVE-2023-43208 and was
+          patched in Mirth Connect version 4.4.1. This module has been tested on versions 4.1.1, 4.3.0 and 4.4.0.
+        },
+        'Author' => [
+          'r00t',
+          'Naveen Sunkavally',
+          'Spencer McIntyre'
+        ],
+        'References' => [
+          ['CVE', '2023-37679'],
+          ['URL', 'https://www.ihteam.net/advisory/mirth-connect/'],
+          ['CVE', '2023-43208'],
+          ['URL', 'https://www.horizon3.ai/nextgen-mirth-connect-remote-code-execution-vulnerability-cve-2023-43208/'],
+          ['URL', 'https://www.horizon3.ai/writeup-for-cve-2023-43208-nextgen-mirth-connect-pre-auth-rce/'],
+        ],
+        'DisclosureDate' => '2023-10-25',
+        'License' => MSF_LICENSE,
+        'Platform' => ['unix', 'linux', 'win'],
+        'Arch' => [ARCH_CMD],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => ['unix', 'linux'],
+              'Arch' => ARCH_CMD
+            }
+          ],
+          [
+            'Windows Command',
+            {
+              'Platform' => 'win',
+              'Arch' => ARCH_CMD,
+              'Payload' => { 'Space' => 8191, 'DisableNops' => true }
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8443,
+          'SSL' => true
+        },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Base path', '/'])
+    ])
+  end
+
+  def check
+    target_version = get_target_version
+    return CheckCode::Unknown unless target_version
+
+    vprint_status("Detected target version: #{target_version}")
+
+    if target_version <= Rex::Version.new('4.3.0')
+      return CheckCode::Appears("Version #{target_version} is affected by CVE-2023-37679.")
+    elsif target_version <= Rex::Version.new('4.4.0')
+      return CheckCode::Appears("Version #{target_version} is affected by CVE-2023-43208.")
+    end
+
+    CheckCode::Safe("Version #{target_version} is not affected.")
+  end
+
+  def get_target_version
+    return @target_version if @target_version
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, 'api/server/version'),
+      'headers' => {
+        'X-Requested-With' => 'OpenAPI'
+      }
+    )
+    return nil unless res&.code == 200
+    return nil unless res.body =~ /(\d+(\.\d+)*)/
```

Comment:
```suggestion
    return nil unless res.body =~ /([0-9][0-9.]+)/
```

at least according to https://github.com/nextgenhealthcare/connect/blob/development/server/src/com/mirth/connect/client/core/Version.java

Reply:
Which version does the current regex not match?

Reply:
Well, the, huh, the, well… I should have went to bed instead of looking at github late at night, sorry for the noise :/